plugins/ecs-serviceconnect/config/netconfig.go

```diff
@@ -0,0 +1,199 @@
+	Mode                             NetworkMode // To be used in future
+}
+
+// netConfigJSON defines the network configuration JSON file format for the ecs-serviceconnect cni plugin.
+type netConfigJSON struct {
+	cniTypes.NetConf
+	IngressConfig []IngressConfigJSONEntry `json:"ingressConfig"`
+	EgressConfig  *EgressConfigJSON        `json:"egressConfig"`
+	NetworkConfig *NetworkMode             `json:"networkMode"`
+	EnableIPv6    bool                     `json:"enableIPv6"`
+}
+
+// IngressConfigJSONEntry defines the ingress network config in JSON format for the ecs-serviceconnect cni plugin.
+type IngressConfigJSONEntry struct {
+	ListenerPort  int `json:"listenerPort"`
+	InterceptPort int `json:"interceptPort,omitempty"`
+}
+
+// EgressConfigJSON defines the egress network config in JSON format for the ecs-serviceconnect cni plugin.
+type EgressConfigJSON struct {
+	ListenerPort int            `json:"listenerPort"`
+	VIP          *VIPConfigJSON `json:"vip"`
+}
+
+// VIPConfigJSON defines the EgressVIP network config in JSON format for the ecs-serviceconnect cni plugin.
+type VIPConfigJSON struct {
+	IPV4CIDR string `json:"ipv4Cidr,omitempty"`
+	IPV6CIDR string `json:"ipv6Cidr,omitempty"`
+}
+
+// New creates a new NetConfig object by parsing the given CNI arguments.
+func New(args *cniSkel.CmdArgs) (*NetConfig, error) {
+	// Parse Json configuration.
+	var config netConfigJSON
+	if err := json.Unmarshal(args.StdinData, &config); err != nil {
+		return nil, fmt.Errorf("failed to parse network config: %v", err)
+	}
+
+	// Validate the configuration.
+	if err := validateConfig(config); err != nil {
+		return nil, err
+	}
+
+	// Parse ingress and construct a listener map
+	ingressRedirectToListenerPortMap := make(map[string]string)
+	for _, s := range config.IngressConfig {
+		if s.InterceptPort != 0 {
+			ingressRedirectToListenerPortMap[strconv.Itoa(s.ListenerPort)] = strconv.Itoa(s.InterceptPort)
+		}
+	}
+	// Parse egress
+	egressPort := 0
+	var egressIPV4CIDR, egressIPV6CIDR string
+	if config.EgressConfig != nil {
+		egressPort = config.EgressConfig.ListenerPort
+		egressIPV4CIDR = config.EgressConfig.VIP.IPV4CIDR
+		egressIPV6CIDR = config.EgressConfig.VIP.IPV6CIDR
+	}
+
+	// Populate NetConfig.
+	netConfig := NetConfig{
+		NetConf:                          config.NetConf,
+		IngressRedirectToListenerPortMap: ingressRedirectToListenerPortMap,
+		EgressPort:                       egressPort,
+		EgressIPV4CIDR:                   egressIPV4CIDR,
+		EgressIPV6CIDR:                   egressIPV6CIDR,
+		IPProtocols:                      getIPProtocols(config),
+		Mode:                             *config.NetworkConfig,
+	}
+	return &netConfig, nil
+}
+
+// validateConfig validates the given network configuration.
+func validateConfig(config netConfigJSON) error {
+	if len(config.IngressConfig) == 0 && config.EgressConfig == nil {
+		return fmt.Errorf("either IngressConfig or EgressConfig must be present")
+	}
+	if config.NetworkConfig == nil {
```

Comment:
Not really. `ValidateNetworkConfig` checks whether the network mode is valid. I will rename `validateNetworkConfig` to `validateNetworkMode` to avoid confusion.